src/data_hub/api.py

```
@@ -817,25 +825,29 @@ def get_dataframe(
         return_type: Whether to return a `pandas.Dataframe` or a dictionary.
     """
 
+    # TODO: list_database_rows and describe_row can be called in parallel
+
     # figure out the rows
     rows = _api.list_database_rows(
         database_row_id=database_id,
         client=client,
     )
 
-    # filter out template rows
-    rows = [
-        row for row in rows if not (hasattr(row, "is_template") and row.is_template)
-    ]
-
     # figure out the column names and ID of the database
     db_row = _api.describe_row(
         row_id=database_id,
         client=client,
     )
-    assert (
-        db_row.type == "database"
-    ), f"Expected database_id: {database_id} to resolve to a database, but instead, it resolved to a {db_row.type}"
```

we shouldn't have an assert here. we should raise the correct exception